src/components/colorscale/calc.js

```diff
@@ -33,7 +33,12 @@ module.exports = function calc(gd, trace, opts) {
             updateStyle(trace._input, containerStr ? (containerStr + '.' + attr) : attr, inputVal);
         }
         else {
-            inputContainer[attr] = inputVal;
+            if(trace._input.type === 'scatter3d') {
+                trace._input[attr] = inputVal;
```

Comment:
Hmm. Looks to me like `containerStr` about isn't getting set properly? What's its value here?

Reply:
The `inputContainer` was `undefined` if I recall correctly.

Reply:
Right , but what about `containerStr`?

From this block above

https://github.com/plotly/plotly.js/blob/3ea5c82de6790a3471fea10774fc50b9c3023ae6/src/components/colorscale/calc.js#L50-L54

this leads be to believe that `containerStr` isn't getting set properly.

Reply:
First it is equal to `line`; but then when it is set to `marker` we get an error.

Reply:
> but then when it is set to `marker` we get an error.

Right, and in this case `Colorscale.calc` should get called in the first place. Something must be not working correctly here:

https://github.com/plotly/plotly.js/blob/3ea5c82de6790a3471fea10774fc50b9c3023ae6/src/traces/scatter/colorscale_calc.js#L27-L34